R/read_builtin.R

```diff
@@ -16,10 +16,12 @@ read_builtin <- function(x, package = NULL) {
   warn_to_error <- function(e) {
     stop(conditionMessage(e), call. = FALSE)
   }
+  check_string(x)
+  check_string(package, optional = TRUE)
   tryCatch(
     warning = function(e) warn_to_error(e),
     expr = {
-      res <- utils::data(list = list(x), package = package, envir = environment(), verbose = FALSE)
+      res <- utils::data(list = x, package = package, envir = environment(), verbose = FALSE)
```

Comment:
I think the original `list(x)` was sort of a mistake. The `list` argument of `utils::data()` is supposed to be a character vector (despite its confusing name).

Reply:
Actually I've done more tests now and this really is the mistake, at least on our end. Just fixing this alone would be enough. But I think we might as well check the inputs and deliver a more specific error message.

Reply:
I find it a little silly that the argument is named `list` but wants a character vector 😛 

Agreed that this was a bug on our end.